mlrun/datastore/azure_blob.py

```diff
@@ -86,12 +83,12 @@ def _convert_key_to_remote_path(self, key):
 
     def upload(self, key, src_path):
         remote_path = self._convert_key_to_remote_path(key)
-        self._filesystem.put_file(src_path, remote_path, overwrite=True)
+        self.get_filesystem().put_file(src_path, remote_path, overwrite=True)
```

Comment:
I think if we're going to replace it everywhere, it makes sense to make it a `@property`.

Reply:
get_filesystem() is a public method in the DataStore class and is overridden in subclasses as part of the public interface. Would it be appropriate to convert it into a @property instead?

Reply:
I don't think it's really meant to be user facing, but if in doubt, consider adding this to the base class:
```python
def get_filesystem(self):
    return self.filesystem
```

Reply:
Let's add it with deprecation warning